pkg/dataobj/internal/dataset/value.go

```diff
@@ -38,6 +38,9 @@ type Value struct {
 	// types.
 	num uint64
 
+	// cap holds the capacity for byte slice pointed to by any, if applicable.
+	cap uint64
```

Comment:
While this change appears small, it increases the size of `Value` from 24 bytes (8 for `num`, 16 for `any`) to 32. That's 33% more memory for all the Value slices and Values themselves. 

Have you taken a look at how much memory comes from Value? If it's a significant, we may want to consider other strategies, such as `any` holding a pointer to some larger struct beyond a `*byte`.

Reply:
I think we might want to do that if we implement Owen's suggestions about not instantiating a value for every column if we don't need all the columns, and then implement datatypes such as that different decoders are required for different columns.
We could probably prevent it by reusing Values by type and only reusing pointer values for other string/byte columns. I don't see any blockers to this so we should be OK with it in the future.